parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -137,6 +137,41 @@ public void setPageSource(PageReadStore pageStore) {
     }
   }
 
+  static class PositionReader implements ParquetValueReader<Long> {
+    private long rowOffsetInCurrentRowGroup = -1;
+    private long rowGroupRowOffsetInFile;
+
+    @Override
+    public Long read(Long reuse) {
+      rowOffsetInCurrentRowGroup = rowOffsetInCurrentRowGroup + 1;
+      return rowGroupRowOffsetInFile + rowOffsetInCurrentRowGroup;
+    }
+
+    @Override
+    public TripleIterator<?> column() {
+      return NullReader.NULL_COLUMN;
+    }
+
+    @Override
+    public List<TripleIterator<?>> columns() {
+      return NullReader.COLUMNS;
+    }
+
+    @Override
+    public void setPageSource(PageReadStore pageStore) {
+    }
+
+    @Override
+    public void setRowOffsetForRowGroup(long rowGroupStartPos) {
+      this.rowGroupRowOffsetInFile = rowGroupStartPos;
+      this.rowOffsetInCurrentRowGroup = -1;
+    }
+  }
+
+  public static ParquetValueReader<Long> position() {
+    return new PositionReader();
+  }
```

Comment:
Can you move this to the top of the file with the other factory methods?

Reply:
Done.